scripts/clangfmt

```diff
@@ -26,28 +25,29 @@ die() {
 check_clang_format_version() {
   cmd="$1"
   [ -e "$(type -P "$cmd")" ] && \
-    "$cmd" --version 2> /dev/null | grep -q "$CLANG_FORMAT_VERSION_STRING"
+    version=$("$cmd" --version | cut -d" " -f3)
+    major=$(echo $version | cut -d. -f1)
+    [ $major -eq $CLANG_FORMAT_VERSION ]
 }
 
 clang_format=
 
 if [ -n "${CLANG_FORMAT_PATH:-}" ]; then
   check_clang_format_version "$CLANG_FORMAT_PATH" || \
-    die "CLANG_FORMAT_PATH does not have required version $CLANG_FORMAT_VERSION" \
+    die "CLANG_FORMAT_PATH must point to version $CLANG_FORMAT_VERSION" \
         "CLANG_FORMAT_PATH points to $CLANG_FORMAT_PATH"
   clang_format="$CLANG_FORMAT_PATH"
 else
-  for cmd in "clang-format-$CLANG_FORMAT_VERSION" clang-format; do
-    if check_clang_format_version "$cmd"; then
-      clang_format="$cmd"
-      break
-    fi
-  done
+  if check_clang_format_version clang-format; then
+    clang_format=clang-format
+  fi
 fi
 
 if [ -z "$clang_format" ]; then
   die "clang-format version $CLANG_FORMAT_VERSION not found" \
-      "Install LLVM version $CLANG_FORMAT_VERSION and rerun."
+      "clang-format version $version found" \
```

Comment:
Maybe let's change it to `clang-format version $CLANG_FORMAT_VERSION expected, but version $version found`?

Reply:
Very good idea, thanks.